### DISCUSSION TITLE
关于麻将强化学习state的问题

### DISCUSSION THREAD

**Discussion by chihiro-zza** [Equim-chan/Mortal#68] [Q&A]
请问一下麻将强化学习的连续状态如何定义。是将`上一轮摸牌时的状态`和`下一轮到自己时候的摸牌状态`，这两个状态定义为一个连续state吗

**Comment by Equim-chan** [comment]
只要是能操作的 state 都会拿出来。

**Reply by chihiro-zza** [reply]
如果像suphx只训练一个丢牌强化模型，在一个agent和三个普通玩家对打时，`摸牌时的状态`和`弃牌后的状态`这两个也算一个连续状态吗，这种连续state对于单人来说是不是没什么意义

**Reply by Equim-chan** [reply]
实际上我没太理解你所说的「连续状态」具体指的是什么，是指邻接的两个状态是什么吗？

「(自家)弃牌后的状态」并不能做任何操作，所以不会当作 state。

**Reply by chihiro-zza** [reply]
在弃牌的强化学习中，一个trajectories对应的state是`摸牌时的状态`,next_state是指`下一轮到自己时候的摸牌状态`吗，reward对应这两个状态间的奖励

**Reply by Equim-chan** [reply]
Mortal 没有像 Suphx 一样区分不同动作类型的模型，而是只用了一个模型来做包括打牌、吃、碰、杠在内的所有动作类型。所以 state 不光是摸牌时的状态，也包括能吃、能碰等时候的状态。

**Reply by chihiro-zza** [reply]
我理解的suphx中的trajectories中state和next_state是正确的吗，对应mortal，state和next_state就是指自家连续的动作，比如说摸牌和下一轮的摸牌，也可以是摸牌和下一轮自家的吃牌

**Reply by Equim-chan** [reply]
是的

**Reply by chihiro-zza** [reply]
感谢解答
